src/app-layout/__tests__/runtime-drawers.test.tsx

```diff
@@ -1386,5 +1387,57 @@ describe('toolbar mode only features', () => {
       globalDrawersWrapper.findCloseButtonByActiveDrawerId(drawerIdWithToggle)!.click();
       expect(onToggle).toHaveBeenCalledWith({ isOpen: false, initiatedByUserAction: true });
     });
+
+    describe('global drawers aria role in overflow menu', () => {
+      const registerGlobalDrawers = (count: number) => {
+        for (let i = 1; i <= count; i++) {
+          awsuiPlugins.appLayout.registerDrawer({
+            ...drawerDefaults,
+            id: `global-drawer${i}`,
+            type: 'global',
+          });
+        }
+      };
+
+      test('assigns correct ARIA roles when mixing global and regular drawers', async () => {
+        registerGlobalDrawers(2);
+        const { wrapper } = await renderComponent(<AppLayout drawers={manyDrawers} />);
+        const buttonDropdown = wrapper.findDrawersOverflowTrigger();
+
+        buttonDropdown!.openDropdown();
+
+        expect(buttonDropdown!.findItemById('global-drawer1')).toBeTruthy();
+        expect(buttonDropdown!.findItemById('global-drawer2')).toBeTruthy();
+
+        const menu = buttonDropdown!.findOpenDropdown()!.find('[role="menu"]')!;
+        // Global drawers should have role=menuitemcheckbox
+        expect(menu.findAll('[role="menuitemcheckbox"]')).toHaveLength(2);
+
+        // Regular drawers should have role=menuitem
+        const menuItems = menu.findAll('[role="menuitem"]');
+        expect(menuItems).toHaveLength(99);
```

Comment:
```suggestion
        expect(menuItems).toHaveLength(manyDrawers.length);
```

Otherwise, if someone updates `manyDrawers`, this test fails